components/support/nimbus-fml/src/workflows.rs

```diff
@@ -5,23 +5,63 @@
 use crate::{
     backends,
     commands::{GenerateExperimenterManifestCmd, GenerateIRCmd, GenerateStructCmd, TargetLanguage},
-    error::Result,
+    error::{FMLError, Result},
     intermediate_representation::FeatureManifest,
     parser::{AboutBlock, Parser},
+    util::loaders::{FileLoader, FilePath},
 };
 use std::path::Path;
 
 #[allow(dead_code)]
 pub(crate) fn generate_struct(cmd: &GenerateStructCmd) -> Result<()> {
-    let ir = load_feature_manifest(&cmd.manifest, cmd.load_from_ir, &cmd.channel)?;
+    let files: FileLoader = TryFrom::try_from(&cmd.loader)?;
+
+    let input = files.file_path(&cmd.manifest)?;
+
+    match (&input, &cmd.output.is_dir()) {
+        (FilePath::Remote(_), _) => generate_struct_single(&files, input, cmd),
+        (FilePath::Local(p), _) if !p.is_dir() => generate_struct_single(&files, input, cmd),
+        (FilePath::Local(d), true) if d.is_dir() => {
+            generate_struct_from_dir(&files, cmd, &cmd.output)
+        }
+        _ => Err(FMLError::CliError(
+            "Cannot generate a single output file from an input directory".to_string(),
+        )),
+    }
+}
+
+fn generate_struct_from_dir(files: &FileLoader, cmd: &GenerateStructCmd, dir: &Path) -> Result<()> {
+    let entries = dir.read_dir()?;
+    for entry in entries {
+        let p = entry?.path();
+        if p.is_dir() {
+            generate_struct_from_dir(files, cmd, &p)?;
+        } else if let Some(nm) = p.file_name().map(|s| s.to_str().unwrap_or_default()) {
+            if nm.ends_with(".fml.yaml") {
+                let path = p.as_path().into();
+                generate_struct_single(files, path, cmd)?;
+            }
+        }
+    }
+    Ok(())
+}
```

Comment:
Look for `.fml.yaml` files here. We might isolate this into a constant.

Reply:
I've tried. Perhaps better minds can come up with someting, eh @tarikeshaq ?

Reply:
If all you are doing is recursively digging for *.fml.yaml files, perhaps a crate like [glob](https://docs.rs/glob/0.3.0/glob/index.html) would be more efficient for this?

Reply:
I added `glob` because it's nice. However, there is a real danger of shell substitution/expansion helpfully being helpful and doing the expansion on the command line, instead of passing the pattern to Rust.